feat: improve checkbox/radio documentation (#5658)
BrightspaceUI/core: components/inputs/docs/input-checkbox.md

# Checkbox Input

Checkboxes are used in forms to toggle an option or preference. They may be grouped to form a set of options.

<!-- docs: demo display:block -->
```html
<script type="module">
  import '@brightspace-ui/core/components/inputs/input-checkbox.js';
  import '@brightspace-ui/core/components/inputs/input-checkbox-group.js';
</script>
<d2l-input-checkbox-group label="Toppings">
  <d2l-input-checkbox label="Ketchup" checked></d2l-input-checkbox>
  <d2l-input-checkbox label="Mustard"></d2l-input-checkbox>
  <d2l-input-checkbox label="Relish"></d2l-input-checkbox>
</d2l-input-checkbox-group>
```

## Best Practices
<!-- docs: start best practices -->
<!-- docs: start dos -->
* Use in a form to indicate an option or preference.
* Use to allow the user to select multiple, independent options from a set.
* When multiple options are presented together, [wrap them in a group](#d2l-input-checkbox-group).

> * Use an indeterminate state to indicate a mixed state where some child items are checked and some are not.

Reviewer: Looks good! 

Can we remove the periods though? We try not to use them for single-sentence bullets (though I think these can still be found in some places).

Author: Yep, done!

* Use as progressive disclosure in forms, so long as users are made aware both visually and non-visually that new options have been made available.
<!-- docs: end dos -->

<!-- docs: start donts -->
* Don't use as a toggle that performs an immediate action, use a Switch component.

> * Don't use for mutually exclusive options, use radio input groups.

Reviewer: Here too, with the period.

We could also link to the radio input page? And the switch page while we're at it.

Author: I've removed the periods. I tried linking to the radio input page but all it did was link to the page in core's repo, not within the Daylight Site. Maybe this isn't possible?

* Don't use labels as “instructions” or “phrases”. Good label: “Visible to Students”. Bad label: (“Check this to make it visible to students”)
* Don't use labels to describe the default or “off” state of the option. As much as possible, use the label to refer to the “on” state. Good label: “Visible”. Bad label: “Hidden”.
<!-- docs: end donts -->
<!-- docs: end best practices -->

## Checkbox Input Group [d2l-input-checkbox-group]

When multiple checkboxes are displayed together, wrap them in a `<d2l-input-checkbox-group>`.

> This will not only provide consistent spacing between checkboxes, but the group also internally renders a `<fieldset>` and `<legend>` with the provided `label`, which gives additional accessibility context. The label can be hidden visually if desired.

Reviewer 2: This ends up debated a lot in these docs. The argument behind not surfacing details is generally in case we change what we're doing internally. This one is likely ok since it's unlikely to change and is teaching a bit about proper checkbox group structure. These details could also instead be moved to the "Accessibility" section below (@geurts not sure if you have thoughts?).

> This will not only provide consistent spacing between checkboxes, but the group also internally renders a `<fieldset>` and `<legend>` with the provided `label`, which gives additional accessibility context. The label can be hidden visually if desired.

Author: The reason I called it out here is so that devs know not to also wrap their checkboxes in a `<d2l-input-fieldset>` (or `<fieldset>`) because we're doing it for them.

<!-- docs: demo code properties name:d2l-input-checkbox-group sandboxTitle:'Checkbox Group' display:block -->
```html
<script type="module">
  import '@brightspace-ui/core/components/inputs/input-checkbox.js';
  import '@brightspace-ui/core/components/inputs/input-checkbox-group.js';
</script>
<d2l-input-checkbox-group label="Toppings">
  <d2l-input-checkbox label="Ketchup"></d2l-input-checkbox>
  <d2l-input-checkbox label="Mustard"></d2l-input-checkbox>
  <d2l-input-checkbox label="Relish"></d2l-input-checkbox>
</d2l-input-checkbox-group>
```

<!-- docs: start hidden content -->
### Properties

| Property | Type | Description |
|---|---|---|
| `label` | String, required | Label for the group of checkboxes |
| `label-hidden` | Boolean | Hides the label visually |
<!-- docs: end hidden content -->

## Checkbox Input [d2l-input-checkbox]

The `<d2l-input-checkbox>` element can be used on its own or as an option within a `<d2l-input-checkbox-group>` collection. It will display a checkbox and optional visible label.

> When provided with a `name`, the checkbox can participate in forms when it is `checked` and enabled. Its value will be `"on"` unless another is provided using the `value` attribute.

Reviewer 3: ```suggestion
When provided with a `name`, the checkbox will participate in forms if it is `checked` and enabled.
```

This just feels more precise. Removed the `value` line because it's already in the properties table, and this is the same as a native checkbox.

<!-- docs: end hidden content -->

## Checkbox Input [d2l-input-checkbox]

The `<d2l-input-checkbox>` element can be used to get a checkbox and optional visible label.
The `<d2l-input-checkbox>` element can be used on its own or as an option within a `<d2l-input-checkbox-group>` collection. It will display a checkbox and optional visible label.

When provided with a `name`, the checkbox can participate in forms when it is `checked` and enabled. Its value will be `"on"` unless another is provided using the `value` attribute.

<!-- docs: demo code properties name:d2l-input-checkbox sandboxTitle:'Checkbox Input' display:block -->
```html
<script type="module">
  import '@brightspace-ui/core/components/inputs/input-checkbox.js';

>   window.addEventListener('load', () => {

Author: This didn't seem particularly helpful and we don't normally provide this kind of code sample.

* `supporting`: Supporting information which will appear below and be aligned with the checkbox.
<!-- docs: end hidden content -->

### Groups of Checkboxes

If multiple checkboxes are displayed together, wrapping them in a `<d2l-input-checkbox-group>` provides consistent spacing between each checkbox.

<!-- docs: demo code -->
```html
<script type="module">
  import '@brightspace-ui/core/components/inputs/input-checkbox.js';
  import '@brightspace-ui/core/components/inputs/input-checkbox-group.js';
</script>
<d2l-input-checkbox-group label="Toppings">
  <d2l-input-checkbox label="Ketchup"></d2l-input-checkbox>
  <d2l-input-checkbox label="Mustard"></d2l-input-checkbox>
  <d2l-input-checkbox label="Relish"></d2l-input-checkbox>
</d2l-input-checkbox-group>
```

> ## Applying styles to native checkboxes

Author: Moved.
